firedrake/parameters.py

```diff
@@ -87,6 +87,9 @@ def set_update_function(self, callable):
 
 parameters["type_check_safe_par_loops"] = False
 
+parameters.add(Parameters("la_compiler"))
+parameters["la_compiler"]["optimise_slate"] = True
+
```

Comment:
Call this `Parameters("slate")` and just `"optimise"` (or "optimise_expressions") ? e.g.` parameters["slate"]["optimise"] = True`

Reply:
I called this "la_compiler" (=linear algebra compiler) for consistency reasons because the tsfc compiler options got the the key "form_compiler".

I don't really mind, but if anything I would do Parameters("tsslac") because slate is just the language and tsslac is the compiler. Just tbc, if we would do the same for tsfc,  the key would be "ufl" that would be odd.

Reply:
I think I would argue for something with slate because that's the name that this part of firedrake is referred to in papers/talks etc... The `form_compiler` thing for tsfc is historical compatibility. Arguably that could be `Parameters("tsfc")` because that's the name people know.

Reply:
Right. "slate_compiler" then. :)

Reply:
👍